src/commands.rs

```diff
@@ -1239,25 +1242,43 @@ fn cmd_config(
     command: &CommandHelper,
     subcommand: &ConfigSubcommand,
 ) -> Result<(), CommandError> {
-    ui.request_pager();
     let settings = command.settings();
     match subcommand {
-        ConfigSubcommand::List { name } => {
-            let raw_values = match name {
-                Some(name) => {
-                    settings
-                        .config()
-                        .get::<config::Value>(name)
-                        .map_err(|e| match e {
-                            config::ConfigError::NotFound { .. } => {
-                                user_error("key not found in config")
-                            }
-                            _ => e.into(),
-                        })?
+        ConfigSubcommand::List {
+            name,
+            include_defaults,
+        } => {
+            let mut values = vec![];
+            for (origin, config) in command.dump_layered_configs() {
+                // Skip built-ins if not included.
+                if origin == ConfigOrigin::Default && !include_defaults {
+                    continue;
                 }
-                None => settings.config().collect()?.into(),
-            };
-            write_config_entry(ui, name.as_deref().unwrap_or(""), raw_values)?;
+                if let Some(name) = name {
+                    // Collect any match into values.
+                    match config.get::<config::Value>(name) {
+                        Ok(val) => values.push(val),
+                        Err(config::ConfigError::NotFound { .. }) => {}
+                        Err(e) => return Err(e.into()),
+                    }
```

Comment:
Felt like there might be a helper to simplify this part better, to convert the `Result<Value>` to a `Result<Option<Value>>` while swallowing the `NotFound` special case or something like that.

Reply:
Yeah, it's annoying, but I couldn't find a library function for that.

Maybe we need to add a helper function or extension trait (like `IoResultExt`.

Reply:
K, as long as there's not an obvious tidy way to do it, I'll resolve this comment for now and keep an eye out for a way to simplify it in future changes.